modules/godot/databags/databag_timer.cpp

```diff
@@ -0,0 +1,70 @@
+	now = p_now;
+}
+
+bool TimersDatabag::is_valid_timer_handle(const godex::TimerHandle p_timer_handle) const {
+	return timers.size() < p_timer_handle.timer_index && timers[p_timer_handle.timer_index].generation == p_timer_handle.generation && timers[p_timer_handle.timer_index].generation > 0;
+}
+
+godex::TimerHandle TimersDatabag::new_timer(const real_t p_end_in_seconds) {
+	ERR_FAIL_COND_MSG(static_cast<double>(now) + p_end_in_seconds * 1'000'000.0 < 0, "Can't make a new timer that ends before 0.");
+	return new_precise_timer(static_cast<uint64_t>(p_end_in_seconds * 1'000'000.0));
+}
+
+godex::TimerHandle TimersDatabag::new_precise_timer(const uint64_t p_end_in_microseconds) {
+	if (destroyed_timers.size() > 0) {
+		godex::TimerHandle finalHandle = destroyed_timers[destroyed_timers.size() - 1];
+		internal_restart_precise_timer(finalHandle, now + p_end_in_microseconds);
+		destroyed_timers.remove_unordered(destroyed_timers.size() - 1);
+		return finalHandle;
+	} else {
+		timers.push_back(godex::Timer(now + p_end_in_microseconds, 1));
+		return godex::TimerHandle(timers.size() - 1, 1);
+	}
+}
+
+void TimersDatabag::restart_timer(const godex::TimerHandle p_timer_handle, const real_t p_end_in_seconds) {
+	ERR_FAIL_COND_MSG(static_cast<double>(now) + p_end_in_seconds * 1'000'000.0 < 0, "Can't make a new timer that ends before 0.");
+	restart_precise_timer(p_timer_handle, static_cast<uint64_t>(p_end_in_seconds * 1'000'000.0));
+}
+
+void TimersDatabag::restart_precise_timer(const godex::TimerHandle p_timer_handle, const uint64_t p_end_in_microseconds) {
+	ERR_FAIL_COND_MSG(!is_valid_timer_handle(p_timer_handle), "The handle" + itos(p_timer_handle.timer_index) + " is not a valid handle.");
+	internal_restart_precise_timer(p_timer_handle, p_end_in_microseconds);
+}
+
+bool TimersDatabag::is_done(const godex::TimerHandle p_timer_handle) const {
+	ERR_FAIL_COND_V_MSG(!is_valid_timer_handle(p_timer_handle), true, "The handle" + itos(p_timer_handle.timer_index) + " is not a valid handle.");
+	return timers[p_timer_handle.timer_index].end_time <= now;
+}
+
+void TimersDatabag::destroy_timer(const godex::TimerHandle p_timer_handle) {
+	if (is_valid_timer_handle(p_timer_handle)) {
+		destroyed_timers.push_back(p_timer_handle);
+		destroyed_timers[destroyed_timers.size() - 1].generation++;
+		timers[p_timer_handle.timer_index].end_time = 0;
+		timers[p_timer_handle.timer_index].generation++;
+	}
+}
+
+real_t TimersDatabag::get_remaining_seconds(const godex::TimerHandle p_timer_handle) const {
+	return (is_done(p_timer_handle) ? -static_cast<double>(now - timers[p_timer_handle.timer_index].end_time) : static_cast<double>(timers[p_timer_handle.timer_index].end_time - now)) / 1'000'000.0;
+}
+
+int64_t TimersDatabag::get_remaining_microseconds(const godex::TimerHandle p_timer_handle) const {
+	return is_done(p_timer_handle) ? -(now - timers[p_timer_handle.timer_index].end_time) : timers[p_timer_handle.timer_index].end_time - now;
```

Comment:
We don't have UnitTests to test this, so I would explicitly add the cast `int64`, to avoid weird things:
```
return is_done(p_timer_handle) ? -int64(now - timers[p_timer_handle.timer_index].end_time) : int64(timers[p_timer_handle.timer_index].end_time - now);
```